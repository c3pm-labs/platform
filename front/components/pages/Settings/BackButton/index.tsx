import {
  makeStyles, Typography,
} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: theme.spacing(3),
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
  containerContent: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    border: 'none',
    backgroundColor: 'rgba(0,0,0,0)',
    padding: 0,
    '&:hover': {
      opacity: 0.8,
      cursor: 'pointer',
    },
    '&:focus': {
      outline: 0,
    },
  },
  backSVG: {
    width: '12px',
    marginRight: theme.spacing(3),
  },
  horizonLine: {
    marginTop: theme.spacing(2),
    display: 'flex',
    width: '100%',
    opacity: 0.4,
    border: 'solid',
    height: '0px',
    borderWidth: '0.1px',
    borderRadius: `${theme.shape.borderRadius}px`,
    borderColor: `${theme.palette.text.disabled}`,
  },
}));

export interface BackButtonProps {
  title: string;
  goBack: () => void;
}

function BackButton(props: BackButtonProps): JSX.Element {
  const { title, goBack } = props;
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <button
        type="button"
        className={classes.containerContent}
        onClick={goBack}
      >
        <img className={classes.backSVG} src="/assets/back.svg" alt="back" />
        <Typography>
          {title}
        </Typography>
      </button>
      <div className={classes.horizonLine} />
    </div>
  );
}

export default BackButton;
