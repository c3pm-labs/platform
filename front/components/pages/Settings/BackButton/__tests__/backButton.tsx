import { render } from '@testing-library/react';

import BackButton from '../index';

test('Back Button', () => {
  const { container } = render(
    <BackButton
      title="titre"
      // eslint-disable-next-line no-console
      goBack={() => console.log('go back')}
    />,
  );

  expect(container).toMatchSnapshot();
});
