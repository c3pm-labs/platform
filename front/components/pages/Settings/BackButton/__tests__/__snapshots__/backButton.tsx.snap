// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Button 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <button
      class="makeStyles-containerContent-2"
      type="button"
    >
      <img
        alt="back"
        class="makeStyles-backSVG-3"
        src="/assets/back.svg"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        titre
      </p>
    </button>
    <div
      class="makeStyles-horizonLine-4"
    />
  </div>
</div>
`;
